File: app/src/main/java/nz/co/codebros/quakesnz/ui/MyMapFragment.java
package nz.co.codebros.quakesnz.ui;

import android.os.Bundle;

import com.google.android.gms.maps.CameraUpdateFactory;
import com.google.android.gms.maps.GoogleMap;
import com.google.android.gms.maps.OnMapReadyCallback;
import com.google.android.gms.maps.SupportMapFragment;
import com.google.android.gms.maps.model.LatLng;
import com.google.android.gms.maps.model.MarkerOptions;

import nz.co.codebros.quakesnz.model.Geometry;

public class MyMapFragment extends SupportMapFragment {
    public static final String ARG_LATITUDE = "arg_latitude";
    public static final String ARG_LONGITUDE = "arg_longitude";

    public static MyMapFragment newInstance(Geometry geometry) {
        LatLng coordinates = geometry.getCoordinates();

        Bundle args = new Bundle();
        args.putDouble(ARG_LATITUDE, coordinates.latitude);
        args.putDouble(ARG_LONGITUDE, coordinates.longitude);

        MyMapFragment f = new MyMapFragment();
        f.setArguments(args);
        return f;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        final double latitude = getArguments().getDouble(ARG_LATITUDE);
        final double longitude = getArguments().getDouble(ARG_LONGITUDE);
        final LatLng location = new LatLng(latitude, longitude);
        getMapAsync(savedInstanceState == null ? new OnMapReadyCallback() {
            @Override
            public void onMapReady(GoogleMap googleMap) {
                googleMap.addMarker(new MarkerOptions().position(location));
                googleMap.moveCamera(CameraUpdateFactory.newLatLngZoom(location, 6));
            }
        } : new OnMapReadyCallback() {
            @Override
            public void onMapReady(GoogleMap googleMap) {
                googleMap.addMarker(new MarkerOptions().position(location));
            }
        });
    }
}

File: app/src/main/java/nz/co/codebros/quakesnz/ui/MyMapFragment.kt
package nz.co.codebros.quakesnz.ui

import android.os.Bundle

import com.google.android.gms.maps.CameraUpdateFactory
import com.google.android.gms.maps.GoogleMap
import com.google.android.gms.maps.OnMapReadyCallback
import com.google.android.gms.maps.SupportMapFragment
import com.google.android.gms.maps.model.LatLng
import com.google.android.gms.maps.model.MarkerOptions

import nz.co.codebros.quakesnz.model.Geometry

class MyMapFragment : SupportMapFragment() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val latitude = arguments.getDouble(ARG_LATITUDE)
        val longitude = arguments.getDouble(ARG_LONGITUDE)
        val location = LatLng(latitude, longitude)
        when (savedInstanceState) {
            null -> getMapAsync { googleMap ->
                googleMap.addMarker(MarkerOptions().position(location))
                googleMap.moveCamera(CameraUpdateFactory.newLatLngZoom(location, 6f))
            }
            else -> getMapAsync { googleMap ->
                googleMap.addMarker(MarkerOptions().position(location))
            }
        }
    }

    companion object {
        val ARG_LATITUDE = "arg_latitude"
        val ARG_LONGITUDE = "arg_longitude"

        fun newInstance(geometry: Geometry): MyMapFragment {
            val coordinates = geometry.coordinates

            val bundle = Bundle()
            bundle.putDouble(ARG_LATITUDE, coordinates.latitude)
            bundle.putDouble(ARG_LONGITUDE, coordinates.longitude)

            val fragment = MyMapFragment()
            fragment.arguments = bundle
            return fragment
        }
    }
}
